src/aioauth/base/database.py

```diff
@@ -26,6 +26,14 @@ async def create_token(self, request: Request, client_id: str, scope: str) -> To
             scope=scope,
             revoked=False,
         )
+        await self.save_token(token)
```

Comment:
I think we could take the token directly from `self.save_token`, in which case we will know for sure that the token was created:

```suggestion
        token = await self.save_token(
            Token(
                client_id=client_id,
                expires_in=request.settings.TOKEN_EXPIRES_IN,
                access_token=generate_token(42),
                refresh_token=generate_token(48),
                issued_at=int(time.time()),
                scope=scope,
                revoked=False,
            )
        )

        return token
```

how do you think?

Reply:
I initially implemented it like that, but I thought:
1. if you fail to save an exception will be thrown
2. there's less coupling `create_*` doesn't depend (much) on `save_*`
3. the token should not mutate in any way when saving, so also no reason to return it

I think it should be:
- create token
- attempt to save (this can be taken as side-effect)
- return token

But if you prefer I'll change it, no problem, is something relatively small.

Reply:
yes I agree. but in this case would be great to replace:

The line:

```python
async def save_token(self, token: Token) -> Token:
```

with

```python
async def save_token(self, token: Token) -> None:
```

in this case, save_token will only be responsible for saving the token (the same with authorization_code).

Reply:
Updated!